rubicon_ml/client/mixin.py

```diff
@@ -326,17 +333,61 @@ def dataframes(self, tags=[], qtype="or"):
             The dataframes previously logged to this client object.
         """
         project_name, experiment_id = self._get_parent_identifiers()
-        dataframes = [
-            client.Dataframe(d, self)
-            for d in self.repository.get_dataframes_metadata(
-                project_name, experiment_id=experiment_id
-            )
-        ]
-
+        if name is not None:
+            dataframes = [
+                client.Dataframe(d, self)
+                for d in self.repository.get_dataframes_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+                if d.name == name
+            ]
+        else:
+            dataframes = [
+                client.Dataframe(d, self)
+                for d in self.repository.get_dataframes_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+            ]
         self._filter_dataframes(dataframes, tags, qtype)
 
         return self._dataframes
 
+    def dataframe(self, name=None, id=None):
+        """
+        Get the dataframe logged to this client object.
+
+        Parameters
+        ----------
+        id : str
+            The id of the artifact to get.
+        name : str
+            The name of the artifact to get.
+        Returns
+        -------
+        rubicon.client.Dataframe
+            The dataframe logged to this project with id `id` or name 'name'.
+        """
+        if (name is None and id is None) or (name is not None and id is not None):
+            raise ValueError("`name` OR `id` required.")
+
+        elif name is not None:
+            dataframes = self.dataframes(name=name)
+            if len(dataframes) == 0:
+                raise RubiconException(f"No dataframe found with name {name}.")
+            elif len(dataframes) > 1:
+                warnings.warn(
+                    f"Multiple dataframes found with name {name}."
+                    " Returning most recently logged."
+                )
+            dataframes = dataframes[-1]
+        else:
+            project_name, experiment_id = self._get_parent_identifiers()
+            dataframes = self.repository.get_dataframe_metadata(
```

Comment:
I think you might need to call this with `client.Dataframe(self.repository.... , self)` to make sure its returning a Dataframe object. Also I realized that you're calling your variables 'dataframes' but maybe changing it to `dataframe` when it's just one `dataframe` object would make it more readable.